tests/data/ref_dimensions.yaml

```diff
@@ -12,6 +12,11 @@ NXarchive(NXobject):
       doc: |
         Testarray with dimensions spelled out
       dim: (10, 30)
+    my_dim_without_dimension(NX_NUMBER):
+      dimensions:
+        doc: |
+          Dim without dimension.
+        dim: []
```

Comment:
Let's think this is the real example then how the `nxdl` get back the `rank="dataRank`. I think the rank should be also in `yaml`. 
This example is really a edge case. Thanks to bring such example (you always hit the edge case :laughing:). This indicates that we must add the `rank` attribute in back and forth conversion.

Reply:
Yeah, I agree that it should somehow keep track of the rank="dataRank" for the backwards conversion to nxdl.

Maybe writing the rank in the nyaml is the only way to go but then we are sort of defeating the ideas of the short-hand notation.